spock-core/src/main/java/spock/util/concurrent/PollingConditions.java

```diff
@@ -182,6 +194,9 @@ public void within(double seconds, Closure<?> conditions) throws InterruptedExce
     }
 
     String msg = String.format("Condition not satisfied after %1.2f seconds and %d attempts", elapsedTime / 1000d, attempts);
+    if (timeoutMessage != null) {
+      msg = String.format("%s with message: %s", msg, GroovyRuntimeUtil.invokeClosure(timeoutMessage, testException));
```

Comment:
I'd leave out the `with message`
```suggestion
      msg = String.format("%s: %s", msg, GroovyRuntimeUtil.invokeClosure(timeoutMessage, testException));
```